File: data/scripts/Random.lua
-- LibreSprite
-- Copyright (C) 2021 LibreSprite contributors

local col = app.pixelColor()
local img = app.activeImage()

math.randomseed(os.time())

for y=0, img.height() - 1 do
  for x=0, img.width() - 1 do
    local c = math.random(256)
    img.putPixel(x, y, col.rgba(c,c,c,255))
  end
end

File: LibreSprite/usr/bin/data/scripts/Random.js
// LibreSprite
// Copyright (C) 2024 LibreSprite contributors

const col = app.pixelColor;
const img = app.activeImage;
const h = img.height;
const w = img.width;

for (var y = 0; y < h; ++y) {
    for (var x = 0; x < w; ++x) {
        const c = Math.random() * 256 >>> 0;
        img.putPixel(x, y, col.rgba(c, c, c, 255))
    }
}
